### DISCUSSION TITLE
Monitoring teleport

### DISCUSSION THREAD

**Discussion by PallaviPalisetti** [gravitational/teleport#15614] [Q&A] [answered]
Hi there,

how to forward or push the teleport kube agent logs to fluentd and elasticsearch. (kubernetes deployment)

Thanks 
Pallavi

**Comment by webvictim** [comment, marked as answer]
If you're talking about audit logs, you'd want to push these centrally from your Teleport cluster itself rather than the connected agents.

We have a Fluentd event handler which can pull events from a Teleport cluster and forward them - take a look at the docs: https://goteleport.com/docs/setup/guides/fluentd/

**Comment by PallaviPalisetti** [comment]
thanks for response. 
Is there any option to monitor health of teleport kube agent when it is down for some reason and how to collect error messages from agent logs??

**Reply by webvictim** [reply]
You can monitor the `teleport-kube-agent` deployments using Kubernetes health/ready probes. The Helm chart already configures these and will automatically restart the pod if it disconnects from the Teleport cluster.

Collecting messages from logs is something that's up to you to do as part of your Kubernetes deployment.
